src/components/DeploymentTagsInput.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <p-tags-input
+    v-model="internalValue"
+    :placeholder="localization.info.addTagPlaceholder"
+    :options="options"
+    :empty-message="localization.info.all"
+  />
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import { useDeployments } from '@/compositions'
+  import { localization } from '@/localization'
+  import { DeploymentsFilter } from '@/models/Filters'
+
+  const props = defineProps<{
+    selected: string[] | null | undefined,
+    filter?: DeploymentsFilter,
+  }>()
+
+  const emits = defineEmits<{
+    (event: 'update:selected', value: string[] | null): void,
+  }>()
+
+  const internalValue = computed({
+    get() {
+      return props.selected ?? null
+    },
+    set(value) {
+      emits('update:selected', value)
+    },
+  })
+
+  const filter = computed<DeploymentsFilter>(() => {
+    return {
+      ...props.filter,
+    }
+  })
```

Comment:
`useDeployments` accepts an undefined ref so rather than a computed for this I'd just do `const { filter } = toRefs(props)`. Little cleaner and more performant.

Reply:
Hm the overload for `useDeployments` isn't quite correct - but ` const { filter = {} } = toRefs<{ filter?: DeploymentsFilter }>(props)` works. I'm not gonna dig into that more tbh